lua/astrocommunity/file-explorer/oil-nvim/init.lua

```diff
@@ -1,5 +1,6 @@
 return {
   "stevearc/oil.nvim",
+  lazy = not vim.startswith(vim.fn.argv()[1] or "", "oil-ssh://"),
```

Comment:
I don't understand why this is here. You're checking of neovim was opened with a file, but `User AstroFile` does that itself. If you want to load the plugin if the filename is empty, I'd just swap the event out instead of having this lazy = vim.fn.... thing, this just makes things more convoluted.
oil-ssh:// etc should trigger `User AstroFile`

Reply:
`User AstroFile` is after `VimEnter` or `UIEnter`.
The same for another pr.

Reply:
Still doesnt explain the usage. Why not just use those events instead?

Reply:
If `VimEnter` or `UIEnter` is used, there is no advantage gained from lazy loading. Therefore, the approach I've identified is to implement lazy loading correctly by using `User AstroFile` for actual files, and disabling lazy loading when dealing with `oil-ssh://`.

Reply:
I’d rather use those events, than the conveluted Logic provided here. Or disable lazy loading if its in the way

Reply:
@meijieru  Also applies to your other PR

Reply:
I agree it's a bit complicated. I learn it from [here](https://github.com/kawre/leetcode.nvim#-recipes). Feel free to close those pr.

Reply:
I think we should just disable lazy loading and call it a day if lazy loading MIGHT cause an issue :)

Reply:
done